src/components/navigation/quickbar.tsx

```diff
@@ -212,78 +214,84 @@ export function Quickbar({ subject, className, placeholder }: QuickbarProps) {
-      let score = 0
-      const preparedQuery = query.toLowerCase().trim()
-      if (entry.titleLower.includes(preparedQuery)) {
-        score += 100
-        if (entry.titleLower.startsWith(preparedQuery)) {
-          score += 10
-        } else if (entry.titleLower.includes(' ' + preparedQuery)) {
-          score += 8
-        }
-      } else {
-        let noHit = 0
-        let kwCount = 0
-        for (const keyword of keywords) {
-          if (keyword) {
-            kwCount++
-            if (entry.titleLower.includes(keyword)) {
-              score += 10
-              continue
-            }
-            let hitContinue = false
-            for (const p of entry.pathLower) {
-              if (p.includes(keyword)) {
-                score += 2
-                hitContinue = true
-                break
-              }
-            }
-            if (hitContinue) continue
-            noHit++
+  return filteredData
+}
+
+export function findResults(data: QuickbarData, query: string) {
+  let results: { entry: QuickbarDataEntry; score: number }[] = []
+
+  const keywords = query.toLowerCase().split(' ')
+
+  for (const entry of data) {
+    let score = 0
+    const preparedQuery = (query.charAt(0) === '/' ? query.slice(1) : query)
+      .toLowerCase()
+      .trim()
+    if (
+      entry.titleLower.includes(preparedQuery) ||
+      preparedQuery === entry.id
+    ) {
+      score += 100
+      if (entry.titleLower.startsWith(preparedQuery)) {
+        score += 10
+      } else if (entry.titleLower.includes(' ' + preparedQuery)) {
+        score += 8
+      } else if (preparedQuery.includes(entry.id)) {
+        score += 8
+      }
+    } else {
+      let noHit = 0
+      let kwCount = 0
+      for (const keyword of keywords) {
+        if (keyword) {
+          kwCount++
+          if (entry.titleLower.includes(keyword)) {
+            score += 10
+            continue
           }
-        }
-        if (kwCount > 0) {
-          if (noHit >= kwCount / 2) {
-            score = 0
-          } else {
-            score *= 1 - noHit / kwCount
+          let hitContinue = false
+          for (const p of entry.pathLower) {
+            if (p.includes(keyword)) {
+              score += 2
+              hitContinue = true
+              break
+            }
           }
+          if (hitContinue) continue
+          noHit++
         }
       }
-      if (score > 0) {
-        score += Math.log10(entry.count)
-        results.push({ entry, score })
-        results.sort((a, b) => b.score - a.score)
-        results = results.slice(0, 7)
+      if (kwCount > 0) {
+        if (noHit >= kwCount / 2) {
+          score = 0
+        } else {
+          score *= 1 - noHit / kwCount
+        }
       }
     }
+    if (score > 0) {
+      score += Math.log10(entry.count)
+      results.push({ entry, score })
+      results.sort((a, b) => b.score - a.score)
+      results = results.slice(0, 7)
+    }
   }
+  return results
```

Comment:
For performance reasons, I believe it's better to do the sorting once at the end (outside of the forof loop) 

```suggestion
     }
   }
  results.sort((a, b) => b.score - a.score)
  results = results.slice(0, 7)
  return results
```